internal/engine/ruletable/rule_table.go

```diff
@@ -48,6 +48,7 @@ type RuleTable struct {
 	// role policies are per-scope, so the maps takes the form `map[scope]map[role][]roles`
 	parentRoles              map[string]map[string][]string
 	parentRoleAncestorsCache map[string]map[string][]string
+	awaitingHealthyIndex     chan struct{}
```

Comment:
I think that an [atomic bool](https://pkg.go.dev/sync/atomic#Bool) would do the trick here? Channels tend to fail in surprising ways and I am a bit wary of using them.